.github/workflows/leverage-cli-test.yml

```diff
@@ -0,0 +1,59 @@
+name: Leverage CLI Test
+
+on:
+  workflow_dispatch:
+    inputs:
+      leverage_branch:
+        description: "Reference to current Leverage branch to test"
+        required: true
+        type: string
+
+jobs:
+  test_leverage:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout base branch
+        uses: actions/checkout@v3
+
+      - name: Clone Leverage
+        run: |
+          printf "[INFO] Cloning Leverage CLI's repo"
+          git clone git@github.com:binbashar/leverage.git
+
+      - name: Build Leverage CLI
+        run: |
+          printf "[INFO] Building Leverage CLI"
+          git checkout ${{ inputs.leverage_branch }}
+          make build
+          pip install -e .
+        working-directory: ./leverage
+
+      - name: Set up credentials
```

Comment:
@angelofenoglio I add an alternative for this step 

[ ] ["Configure AWS Credentials" Action For GitHub Actions](https://github.com/aws-actions/configure-aws-credentials)